smoketests/__init__.py

```diff
@@ -237,11 +237,14 @@ def stderr_task():
             sys.stderr.writelines(proc.stderr)
         threading.Thread(target=stderr_task).start()
 
-        init_update = proc.stdout.readline()
-        code = proc.poll()
-        if code:
-            raise subprocess.CalledProcessError(code, fake_args)
-        print("initial update:", init_update)
+        init_update = proc.stdout.readline().strip()
+        if init_update:
+            print("initial update:", init_update)
+        else:
+            code = proc.poll()
+            if code:
+                raise subprocess.CalledProcessError(code, fake_args)
+            print("no inital update, but no error code either")
```

Comment:
to double-check - is it intended that we trigger this _both_ if the `code` is `None` _and_` if the `code` is `0`?

Reply:
Correct. Maybe it should check for 0, but it doesn't make much of a difference anyway.

Reply:
Oh - it should wait. I see the potential race condition here.

Reply:
Yeah I think that's right. I suspect it's basically when the process hasn't quite exited yet, but I was unable to reproduce locally so I couldn't really narrow done.